Split calc package out into an independent package

stvstnfrd opened the pull request:
This is part of a larger effort to split packages out into independent,
composable, PyPI installable Python packages.

While this package itself isn't monumental, it represents the bottom of
the dependecy tree and can be split out before any other packages.

This depends on the work here: https://github.com/stvstnfrd/openedx-calc/pull/1

edx-webhook commented:
Thanks for the pull request, @stvstnfrd! I've created [OSPR-3265](https://openedx.atlassian.net/browse/OSPR-3265) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

stvstnfrd commented on requirements/edx/local.in:
```diff
@@ -1,6 +1,6 @@
 # Python libraries to install that are local to the edx-platform repo
 -e .
--e common/lib/calc
+-e git+https://github.com/stvstnfrd/openedx-calc.git@tests#egg=calc==0.2
```
I suspect this should live somewhere else ;)

stvstnfrd commented:
CC: @ormsbee @nasthagiri

natabene commented:
@stvstnfrd Thank you for your contribution. Please let me know once all checks are green and this is ready for review.

stvstnfrd commented:
CC: @feanil @jmbowman

jolyonb commented:
Make sure that you're careful with the codejail code, which imports `calc` for use in user scripts. I believe this is not presently tested in the unit tests.

feanil commented:
jenkins run all

feanil commented:
jenkins python

feanil commented:
jenkins run python

1anre commented:
Please remove me from this thread.

On 19 Apr 2019, 1:38 AM +0100, edx-status-bot <notifications@github.com>, wrote:
> Your PR has finished running tests. The following contexts failed:
>
> • jenkins/python
> • jenkins/bokchoy
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

natabene commented:
> Please remove me from this thread.
> […](#)
> On 19 Apr 2019, 1:38 AM +0100, edx-status-bot ***@***.***>, wrote: Your PR has finished running tests. The following contexts failed: • jenkins/python • jenkins/bokchoy — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

@1anre You can probably remove yourself by clicking "Unsubscribe" in this PR.

nedbat commented:
jenkins run python 
# Because this failed: https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/4302/

feanil commented:
jenkins run bokchoy

feanil commented:
jenkins run python

feanil commented:
https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/4316/ passed on this PR but never updated github for some reason.

feanil commented:
@stvstnfrd looks like you got it to green!  Do you want to PR your changes from your test branch to master?  Long term, I'm trying to figure out where we want this repo to end up.

feanil commented:
I made https://github.com/edx/openedx-calc and gave you access to it if you want to push this there.

stvstnfrd commented:
Thanks @feanil!
Yeah, I was missing a few commits/refactors that had been added since I first split this out.
I re-ran my subtree script to split out the commits again, and rebased my cleanup on top.

I'll push to the new repo later today!
Thanks for helping to push this forward!

Once we fully button up this one,
I'll move on to `common/lib/chem` next :)

feanil commented on requirements/edx-sandbox/base.in:
```diff
@@ -15,11 +15,11 @@ numpy==1.6.2                        # Numeric array processing utilities; used b
 pyparsing==2.2.0                    # Python Parsing module
 scipy==0.14.0                       # Math, science, and engineering library
 sympy==0.7.1                        # Symbolic math library
+-e git+https://github.com/edx/openedx-calc.git@stv/tests#egg=calc==0.4
```
@stvstnfrd  BTW, I tried this same build without the editable flag here: https://github.com/edx/edx-platform/pull/20279 and it also succeeded so I don't think you need the `-e` on this.

stvstnfrd replied:
Cool, thanks @feanil , I can make that change; I guess I was still cargo-culting..

I was actually hoping we'd be publishing the new package to PyPI.
Assuming so, I'll update again, presumably `openedx-calc`?

feanil replied:
yea, I think we can land your current PR and then do a follow up to get it publishing to PyPI.  Do you want to wait on that before landing this one?

stvstnfrd replied:
Cool, updated to remove `-e` flag.
Note: will squash prior to merge.

stvstnfrd replied:
And that's fine.. just let me know when it's ready to go on PyPI later.

stvstnfrd commented:
@feanil I think the last thing this needs is to update the URL to pin the specific SHA (which I'll do once we merge `openedx-calc` to master) and then to squash these commits.

The last thing I'm blocked on in `openedx-calc` is to enable TravisCI integration, to ensure we still have a green build. Then we can merge there.

jmbowman commented:
I just enabled Travis CI integration on openedx-calc, that should be good to go.

natabene commented:
@ormsbee Looks like this is ready for your review.

nasthagiri commented:
@natabene I believe @feanil is looking at this.  He's away at a conference early this week.

stvstnfrd commented:
Thanks for the test suite, @jmbowman . It ran smoothly.

The changes have now merged to master on `openedx-calc`!

I've gone ahead and updated the requirement to pin to the current commit SHA from `openedx-calc:master` and squashed down commits, so I think this is ready, once @feanil is back in town :)

Thanks everyone!

feanil commented:
jenkins run a11y
jenkins run python

feanil commented:
@stvstnfrd it looks like this is good for a squash and merge.

stvstnfrd commented:
@feanil I had actually intended for those two commits to stand discretely;
- one to enable the new nepo
- another to remove the old code

stvstnfrd commented:
@feanil I'll gladly merge this myself, but

![303gsr](https://user-images.githubusercontent.com/7300975/57158506-7496b080-6d98-11e9-84c4-86ee9733339f.jpg)

edx-webhook commented:
@stvstnfrd 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/20101&entry.1484655318=https://github.com/stvstnfrd&entry.752974735=2019-03-29+19:16&entry.1917517419=2019-05-03+19:59&entry.2133058324=Yes) so we can improve your experience in the future.
